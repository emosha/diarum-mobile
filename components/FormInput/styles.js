import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 20,
    marginVertical: 0,
  },
  formIcon: {
    color: '#5D59CB',
    marginRight: 10,
  },
  input: {
    color: '#333',
    fontSize: 20,
  },
  inputContainer: {
    borderColor: '#5D59CB',
    borderBottomWidth: 2,
  },
  errorInput: {
    borderColor: 'red',
  },
  containerStyle: {
    marginBottom: 25,
    height: 65,
  },
  errorStyle: {
    color: 'red',
  },
});

export default styles;
