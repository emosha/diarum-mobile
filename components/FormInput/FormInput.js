import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import Icon from 'react-native-vector-icons/AntDesign';
import FontAIcon from 'react-native-vector-icons/FontAwesome';
import { Input } from 'react-native-elements';

import styles from './styles';

const FormInput = (props) => {
  const error = props.touched && props.error;

  return (
      <View style={styles.container}>
        <Input
          ref={props.inputRef}
          placeholder={props.placeholder}
          inputStyle={styles.input}
          inputContainerStyle={{ ...styles.inputContainer, ...(error && styles.errorInput) }}
          shake={true}
          textContentType={props.type}
          keyboardType={props.keyboardType}
          secureTextEntry={props.secureTextEntry}
          onBlur={this.onBlur}
          onChangeText={props.onChangeText}
          errorMessage={error ? props.error : null}
          containerStyle={styles.containerStyle}
          enablesReturnKeyAutomatically
          leftIcon={
            <Icon
              name={props.icon}
              size={24}
              style={styles.formIcon}
            />
          }
          rightIcon={
            props.iconRight
            && <FontAIcon
              name={props.iconRight}
              size={24}
              style={styles.formIcon}
              backgroundColor="transparent"
              onPress={props.togglePassword}
            />
          }
          blurOnSubmit={false}
          {...props}
        />
      </View>
  );
};

FormInput.defaultProps = {
  error: null,
  touched: null,
  disabled: false,
  secureTextEntry: false,
  keyboardType: 'default',
};

FormInput.propTypes = {
  inputRef: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
  error: PropTypes.string,
  touched: PropTypes.bool,
  icon: PropTypes.string.isRequired,
  iconRight: PropTypes.string,
  type: PropTypes.string.isRequired,
  keyboardType: PropTypes.string,
  placeholder: PropTypes.string.isRequired,
  onBlur: PropTypes.func.isRequired,
  onChangeText: PropTypes.func.isRequired,
  returnKeyType: PropTypes.string.isRequired,
  secureTextEntry: PropTypes.bool,
  togglePassword: PropTypes.func,
};

export default FormInput;
