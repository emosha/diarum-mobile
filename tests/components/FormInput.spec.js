import React from 'react';
import FormInput from '../../components/FormInput/FormInput';

describe('FormInput', () => {
  let Component;
  const props = {
    error: null,
    touched: false,
    icon: 'key',
    type: 'name',
    placeholder: 'Name',
    onBlur: jest.fn(),
    onChangeText: jest.fn(),
    returnKeyType: 'next',
    inputRef: null,
  };

  beforeAll(() => {
    Component = mount(<FormInput {...props} />);
  });

  it('renders correctly', () => {
    expect(Component).toMatchSnapshot();
  });

  it('does not show error message when input has not been touched but has an error', () => {
    const ComponentWithError = mount(<FormInput {...props} touched={false} error="an error" />);

    expect(ComponentWithError.find('Text').at(1).length).toBeFalsy();
  });

  it('shows error message when input has been touched and has an error', () => {
    const ComponentWithError = mount(<FormInput {...props} touched error="an error" />);

    expect(ComponentWithError.find('Text').at(1).length).toBeTruthy();
    expect(ComponentWithError.find('Text').at(1).text()).toEqual('an error');
  });
});
