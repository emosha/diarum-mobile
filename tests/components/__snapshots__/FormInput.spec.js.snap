// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput renders correctly 1`] = `
<FormInput
  disabled={false}
  error={null}
  icon="key"
  inputRef={null}
  keyboardType="default"
  onBlur={[MockFunction]}
  onChangeText={[MockFunction]}
  placeholder="Name"
  returnKeyType="next"
  secureTextEntry={false}
  touched={false}
  type="name"
>
  <Component
    style={
      Object {
        "marginHorizontal": 20,
        "marginVertical": 0,
      }
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 20,
          "marginVertical": 0,
        }
      }
    >
      <ForwardRef(Themed.Input)
        blurOnSubmit={false}
        containerStyle={
          Object {
            "height": 60,
            "marginBottom": 25,
          }
        }
        disabled={false}
        enablesReturnKeyAutomatically={true}
        error={null}
        errorMessage={null}
        icon="key"
        inputContainerStyle={
          Object {
            "borderBottomWidth": 2,
            "borderColor": "#5D59CB",
          }
        }
        inputRef={null}
        inputStyle={
          Object {
            "color": "#333",
            "fontSize": 20,
          }
        }
        keyboardType="default"
        leftIcon={
          <Icon
            allowFontScaling={false}
            name="key"
            size={24}
            style={
              Object {
                "color": "#5D59CB",
                "marginRight": 10,
              }
            }
          />
        }
        onBlur={[MockFunction]}
        onChangeText={[MockFunction]}
        placeholder="Name"
        returnKeyType="next"
        secureTextEntry={false}
        shake={true}
        textContentType="name"
        touched={false}
        type="name"
      >
        <ThemedComponent
          blurOnSubmit={false}
          containerStyle={
            Object {
              "height": 60,
              "marginBottom": 25,
            }
          }
          disabled={false}
          enablesReturnKeyAutomatically={true}
          error={null}
          errorMessage={null}
          forwardedRef={null}
          icon="key"
          inputContainerStyle={
            Object {
              "borderBottomWidth": 2,
              "borderColor": "#5D59CB",
            }
          }
          inputRef={null}
          inputStyle={
            Object {
              "color": "#333",
              "fontSize": 20,
            }
          }
          keyboardType="default"
          leftIcon={
            <Icon
              allowFontScaling={false}
              name="key"
              size={24}
              style={
                Object {
                  "color": "#5D59CB",
                  "marginRight": 10,
                }
              }
            />
          }
          onBlur={[MockFunction]}
          onChangeText={[MockFunction]}
          placeholder="Name"
          returnKeyType="next"
          secureTextEntry={false}
          shake={true}
          textContentType="name"
          touched={false}
          type="name"
        >
          <Input
            blurOnSubmit={false}
            containerStyle={
              Object {
                "height": 60,
                "marginBottom": 25,
              }
            }
            disabled={false}
            enablesReturnKeyAutomatically={true}
            error={null}
            errorMessage={null}
            icon="key"
            inputContainerStyle={
              Object {
                "borderBottomWidth": 2,
                "borderColor": "#5D59CB",
              }
            }
            inputRef={null}
            inputStyle={
              Object {
                "color": "#333",
                "fontSize": 20,
              }
            }
            keyboardType="default"
            leftIcon={
              <Icon
                allowFontScaling={false}
                name="key"
                size={24}
                style={
                  Object {
                    "color": "#5D59CB",
                    "marginRight": 10,
                  }
                }
              />
            }
            onBlur={[MockFunction]}
            onChangeText={[MockFunction]}
            placeholder="Name"
            returnKeyType="next"
            secureTextEntry={false}
            shake={true}
            textContentType="name"
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
            touched={false}
            type="name"
          >
            <Component
              style={
                Object {
                  "height": 60,
                  "marginBottom": 25,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 60,
                    "marginBottom": 25,
                    "paddingHorizontal": 10,
                    "width": "100%",
                  }
                }
              >
                <AnimatedComponent
                  style={
                    Object {
                      "alignItems": "center",
                      "borderBottomWidth": 2,
                      "borderColor": "#5D59CB",
                      "flexDirection": "row",
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <Component
                    style={
                      Object {
                        "alignItems": "center",
                        "borderBottomWidth": 2,
                        "borderColor": "#5D59CB",
                        "flexDirection": "row",
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                        ],
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "borderBottomWidth": 2,
                          "borderColor": "#5D59CB",
                          "flexDirection": "row",
                          "transform": Array [
                            Object {
                              "translateX": 0,
                            },
                          ],
                        }
                      }
                    >
                      <Component
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 40,
                            "justifyContent": "center",
                            "marginLeft": 15,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "height": 40,
                              "justifyContent": "center",
                              "marginLeft": 15,
                            }
                          }
                        >
                          <Icon
                            allowFontScaling={false}
                            name="key"
                            size={24}
                            style={
                              Object {
                                "color": "#5D59CB",
                                "marginRight": 10,
                              }
                            }
                          >
                            <Component>
                              <Text />
                            </Component>
                          </Icon>
                        </View>
                      </Component>
                      <TextInput
                        allowFontScaling={true}
                        blurOnSubmit={false}
                        disabled={false}
                        enablesReturnKeyAutomatically={true}
                        error={null}
                        icon="key"
                        inputRef={null}
                        keyboardType="default"
                        onBlur={[MockFunction]}
                        onChangeText={[MockFunction]}
                        placeholder="Name"
                        returnKeyType="next"
                        secureTextEntry={false}
                        shake={true}
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#333",
                            "flex": 1,
                            "fontSize": 20,
                            "minHeight": 40,
                          }
                        }
                        testID="RNE__Input__text-input"
                        textContentType="name"
                        touched={false}
                        type="name"
                        underlineColorAndroid="transparent"
                      >
                        <TextInput
                          allowFontScaling={true}
                          blurOnSubmit={false}
                          disabled={false}
                          enablesReturnKeyAutomatically={true}
                          error={null}
                          icon="key"
                          inputRef={null}
                          keyboardType="default"
                          onBlur={[MockFunction]}
                          onChangeText={[MockFunction]}
                          placeholder="Name"
                          returnKeyType="next"
                          secureTextEntry={false}
                          shake={true}
                          style={
                            Object {
                              "alignSelf": "center",
                              "color": "#333",
                              "flex": 1,
                              "fontSize": 20,
                              "minHeight": 40,
                            }
                          }
                          testID="RNE__Input__text-input"
                          textContentType="name"
                          touched={false}
                          type="name"
                          underlineColorAndroid="transparent"
                        />
                      </TextInput>
                    </View>
                  </Component>
                </AnimatedComponent>
              </View>
            </Component>
          </Input>
        </ThemedComponent>
      </ForwardRef(Themed.Input)>
    </View>
  </Component>
</FormInput>
`;
